util/testutil/tar.go

```diff
@@ -122,6 +126,58 @@ func BuildTarGz(ents []TarEntry, compressionLevel int, opts ...BuildTarOption) i
 	return pr
 }
 
+// BuildTarZstd builds a tar blob with zstd compression.
```

Comment:
Do we need this? At least not until we have zstd decompression right?

Reply:
`toc` builder is suppose to work with different compression, and the implementation in this PR supports both gzip and zstd. This PR also has unittest for `toc` builder with zstd.